pkg/securelaunch/helpers.go

```diff
@@ -48,71 +67,230 @@ func WriteToFile(data []byte, dst, defFileName string) (string, error) {
 		Debug("New target=%s", target)
 	}
 
-	Debug("target=%s", target)
+	Debug("WriteToFile: target=%s", target)
 	err = ioutil.WriteFile(target, data, 0644)
 	if err != nil {
 		return "", fmt.Errorf("failed to write date to file =%s, err=%v", target, err)
 	}
-	Debug("WriteToFile exit w success data written to target=%s", target)
+	Debug("WriteToFile: exit w success data written to target=%s", target)
 	return target, nil
 }
 
-/*
- * GetMountedFilePath returns a file path corresponding to a <device_identifier>:<path> user input format.
- * <device_identifier> may be a Linux block device identifier like sda or a FS UUID.
- *
- * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
- */
-func GetMountedFilePath(inputVal string, flags uintptr) (string, string, error) {
-	s := strings.Split(inputVal, ":")
-	if len(s) != 2 {
-		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
+// persist writes data to targetPath.
+// targetPath is of form sda:/boot/cpuid.txt
+func persist(data []byte, targetPath string, defaultFile string) error {
+
+	filePath, r := GetMountedFilePath(targetPath, 0) // 0 is flag for rw mount option
+	if r != nil {
+		return fmt.Errorf("persist: err: input %s could NOT be located, err=%v", targetPath, r)
+	}
+
+	dst := filePath // /tmp/boot-733276578/cpuid
+
+	target, err := WriteToFile(data, dst, defaultFile)
+	if err != nil {
+		log.Printf("persist: err=%s", err)
```

Comment:
I would suggest removing the print; whatever called it can print

Reply:
I have followed this printing logic throughout the sluinit code where in i print the error  message if a function error exits. I would prefer to keep it as-is. if its okay.

But if you really want me to change this, maybe we can do it in the next PR, so that I can consistently change this throughout the sluinit/securelaunch pkg to maintain consistency...
otherwise it would be chaotic to have one logic in one place and different in others..